sdk/storage/Azure.Storage.Blobs.DataMovement/src/BlobScanner.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Specialized;
+
+namespace Azure.Storage.Blobs.DataMovement
+{
+    /// <summary>
+    /// AzureContainerScanner class.
+    /// </summary>
+    public class BlobScanner
+    {
+        private readonly BlobContainerClient _containerClient;
+
+        /// <summary>
+        /// Constuctor for AzureContainerScanner.
+        /// </summary>
+        /// <param name="client"></param>
+        public BlobScanner(BlobContainerClient client)
+        {
+            // Hacky deep-clone to prevent client changes mid-operation
+            _containerClient = client.GetParentBlobServiceClient().GetBlobContainerClient(client.Name);
+        }
+
+        /// <summary>
+        /// Constuctor for AzureContainerScanner.
+        /// </summary>
+        /// <returns>Async enumerable list containing the name of all blobs in the container.</returns>
+        public async IAsyncEnumerable<string> Scan()
+        {
+            // Return the names of all blobs inside the container
+            await foreach (Models.BlobItem blob in _containerClient.GetBlobsAsync().ConfigureAwait(false))
```

Comment:
Anything wrong here with the method in general? Ends up working after making it like this to get past errors, but felt a bit shifty to me. Could be fine though.

Reply:
Hm you don't need the `ConfigureAwait`, but I'm assuming without that it throws an error?

What's the error that gets thrown exactly?

Reply:
Without the ConfigureAwait, I get this error and can't compile:
`AZC0100	ConfigureAwait(false) must be used.	Azure.Storage.Blobs.DataMovement`

I had this all set up synchronously before, returning an IEnumerable and running foreach (not await foreach). That compiled, but it would throw an error that said GetBlobs needed to be called in an async context, so I switched it over.